src/Infrastructure/SshFileFetcher.php

```diff
@@ -24,32 +23,41 @@ class SshFileFetcher implements FileFetcher
      */
     public function fetch(PimConnection $connection, string $filePath, bool $withLocalCopy): string
     {
-        $sftp = $this->createSftpConnection($connection);
+        $ssh = new Ssh($connection->getHost(), $connection->getPort());
+        $sshConnection = $ssh->connect();
+        $ssh->login($connection->getUsername());
 
         $pathInfo = pathinfo($filePath);
         $fileName = $pathInfo['basename'];
 
-        $subList = $sftp->nlist($pathInfo['dirname']);
+        if (!$sftp = ssh2_sftp($sshConnection)) {
```

Comment:
why the object ssh dos not embed the sftp manipulation? ;)

Reply:
Don't worry guys I'll remove sftp use 🏑

Reply:
Cool :)